### PR TITLE
test: merge unit file related tests into TEST-23-UNIT-FILE

### PR DESCRIPTION
**Pull request by mrc0mmand**
Rename TEST-23-TYPE-EXEC to TEST-23-UNIT-FILE and merge it with following tests:
  - TEST-37-RUNTIMEDIRECTORYPRESERV
  - TEST-40-EXEC-COMMAND-EX
  - TEST-41-ONESHOT-RESTART
  - TEST-42-EXECSTOPPOST
  - TEST-57-ONSUCCESS-UPHOLD

### CODE REVIEW

**Comment by bluca** [comment]
Is there any functional change besides merging into one?

**Comment by mrc0mmand** [comment]
> Is there any functional change besides merging into one?

Not really, but one of the tests was blacklisted in Ubuntu CI (on arm64) but I don't know why, so I'm curious if it's going to fail.

**Comment by mrc0mmand** [comment]
Looks like everything works as intended, nice, so PTAL I guess!

**Review by bluca — APPROVED** [review]

File: test/units/testsuite-23.ExecStopPost.sh [hunk 1]
```diff
@@ -100,7 +102,3 @@ test -f /run/idle1
 test -f /run/idle2
 
 systemd-analyze log-level info
```
Comment (yuwata) [inline test/units/testsuite-23.ExecStopPost.sh:104]: Please drop `systemd-analyze log-level` or `systemctl log-level` from subtests, but call it from the main script. The subtest scripts are now running in parallel, hence, we may not get expected logs.

**Review by yuwata — CHANGES_REQUESTED** [review]
Looks nice!

File: test/units/testsuite-23.ExecStopPost.sh (hunk 1 above)
Comment (mrc0mmand) [reply test/units/testsuite-23.ExecStopPost.sh:104]: The subtests are still serialized - the async shenanigans are there just to make traps work, but we still wait for each subtest to finish before running the next one.

Comment (yuwata) [reply test/units/testsuite-23.ExecStopPost.sh:104]: Ah, indeed. I overlooked the comments in the main script. Interesting.

**Review by yuwata — APPROVED** [review]

**Pull request merged**